### PR TITLE
OBPIH-1807 Fix issue with adjusting inventory on pick page

### PR DESCRIPTION
**Pull request by DominikaHatala**

### CODE REVIEW

File: src/js/components/stock-movement-wizard/PickPage.jsx [hunk 1]
```diff
@@ -268,7 +268,7 @@ class PickPage extends Component {
   }
 
   fetchPickPageItems() {
```
Comment (pmuchowski) [inline src/js/components/stock-movement-wizard/PickPage.jsx]: I think you should rename this method

File: src/js/components/stock-movement-wizard/PickPage.jsx [hunk 2]
```diff
@@ -268,7 +268,7 @@ class PickPage extends Component {
...
-    apiClient.get(`/openboxes/api/stockMovements/${this.state.values.stockMovementId}?stepNumber=4`)
+    apiClient.post(`/openboxes/api/stockMovements/${this.state.values.stockMovementId}/updateAdjustedItems`)
```
Comment (pmuchowski) [inline src/js/components/stock-movement-wizard/PickPage.jsx]: the stockAdjustments endpoint will return the list of adjusted items, you can use it and send it to the updateAdjustedItems and make the update only for this stock movement items

File: grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy [hunk 3]
```diff
@@ -223,6 +223,14 @@ class StockMovementApiController {
         render ([data:stockMovement] as JSON)
     }
 
+    def updateAdjustedItems = {
+        StockMovement stockMovement = stockMovementService.getStockMovement(params.id, "4")
```
Comment (pmuchowski) [inline grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy]: it should take as a parameter the list of adjusted items

File: grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy [hunk 4]
```diff
@@ -1049,6 +1049,51 @@ class StockMovementService {
         }
     }
 
+    void updateAdjustedItems(StockMovement stockMovement) {
+        if (stockMovement?.pickPage?.pickPageItems) {
```
Comment (pmuchowski) [inline grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy]: you could use this list here and find only those stock movement items that uses the adjusted products

File: grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy [hunk 5]
```diff
@@ -1049,6 +1049,51 @@ class StockMovementService {
...
+            stockMovement.pickPage.pickPageItems.each { PickPageItem pickPageItem ->
+                updateAdjustment(pickPageItem)
+            }
+        }
+    }
+
+    void updateAdjustedItems(PickPageItem pickPageItem) {
+        RequisitionItem requisitionItem = pickPageItem.requisitionItem
```
Comment (pmuchowski) [inline grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy]: in this method then you will need only to remove the picklist items and shipment items for the stock movement item and create missing picklist and shipment items (to autopick after the quantity changes)

**Comment by jmiranda** [comment]
@DominikaHatala please fix merge conflicts

**Comment by DominikaHatala** [comment]
@jmiranda done

**Pull request merged**